app/Http/Middleware/HMACAuthorization.php

```diff
@@ -170,7 +170,7 @@ protected function validateIPAccess()
      */
     protected function validateContents()
     {
-        if (base64_decode($this->hash()) !== $this->generateSignature()) {
+        if(!hash_equals(base64_decode($this->hash()),$this->generateSignature())){
```

Comment:
As a general code standard, we have a space between `if` and the parenthesis, `!` and functions, as well as after commas.

```
if (! hash_equals(..., ...))
```